packages/mui-base/src/unstable_useNumberInput/useNumberInput.ts

```diff
@@ -104,12 +106,12 @@ export function useNumberInput(parameters: UseNumberInputParameters): UseNumberI
     }
   }, [formControlContext, disabledProp, focused, onBlur]);
 
-  const handleFocus =
-    (otherHandlers: Record<string, React.EventHandler<any> | undefined>) =>
-    (event: React.FocusEvent<HTMLInputElement>) => {
+  const createHandleFocus =
+    (otherHandlers: Partial<EventHandlers>) =>
+    (event: React.FocusEvent<HTMLInputElement> & MuiCancellableEvent) => {
       otherHandlers.onFocus?.(event);
 
-      if (event.defaultPrevented) {
+      if (event.defaultMuiPrevented || event.defaultPrevented) {
```

Comment:
Should we standardize `defaultMuiPrevented` for all the hooks for the stable release?

Reply:
We should standardize, but I don't think it's a release blocker.

Reply:
Yup, we should. Plus, many hooks don't check any of these, so it should be introduced universally. I think this is simple enough for the community to help us.